From a5a2dc1026482a6981e4c6c0b156ccdce7809ba1 Mon Sep 17 00:00:00 2001
From: Gabor Juhos <juhosg@openwrt.org>
Date: Thu, 16 Mar 2023 10:21:21 +0530
Subject: [PATCH 016/281] OpenWrt:
 240-remove-unsane-filenames-from-deps_initramfs-list.patch

usr: sanitize deps_initramfs list

If any filename in the intramfs dependency
list contains a colon, that causes a kernel
build error like this:

/devel/openwrt/build_dir/linux-ar71xx_generic/linux-3.6.6/usr/Makefile:58: *** multiple target patterns.  Stop.
make[5]: *** [usr] Error 2

Fix it by removing such filenames from the
deps_initramfs list.

Signed-off-by: Gabor Juhos <juhosg@openwrt.org>
Signed-off-by: Felix Fietkau <nbd@nbd.name>

Change-Id: Ie6506b1fa1c25451745fbe500307f935a5e3f3e2
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
---
 usr/Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/usr/Makefile b/usr/Makefile
index 59d9e8b07a01..5a8a00cf8580 100644
--- a/usr/Makefile
+++ b/usr/Makefile
@@ -56,6 +56,8 @@ hostprogs := gen_init_cpio
 # The dependency list is generated by gen_initramfs.sh -l
 -include $(obj)/.initramfs_data.cpio.d
 
+deps_initramfs := $(foreach v,$(deps_initramfs),$(if $(findstring :,$(v)),,$(v)))
+
 # do not try to update files included in initramfs
 $(deps_initramfs): ;
 
-- 
2.17.1

